lib/glific/wa_managed_phones.ex

```diff
@@ -183,4 +184,48 @@ defmodule Glific.WAManagedPhones do
   @spec has_any_phones(list()) :: {:ok, list()} | {:error, String.t()}
   defp has_any_phones([]), do: {:error, "No active phones available"}
   defp has_any_phones(wa_managed_phones), do: {:ok, wa_managed_phones}
+
+  @doc """
+  add the phone status
+  """
+  @spec status(String.t(), non_neg_integer()) :: {:ok, WAManagedPhone} | {:error, String.t()}
+  def status(new_status, phone_id) do
+    organization_id = Repo.get_organization_id()
+
+    phone =
+      from(p in WAManagedPhone,
+        where: p.phone_id == ^phone_id
+      )
+      |> Repo.one()
+
+    case phone do
+      nil ->
+        {:error, "Phone ID not found"}
+
+      _phone ->
+        case phone
```

Comment:
Can we use `with` instead of nested `case` here, since it will avoid nesting.

As mentioned in the above comment, we can use the existing functions for checking if phone exist or not.